counter/src/Main.purs

```diff
@@ -18,8 +23,28 @@ main :: Effect Unit
 main = do
   el <- getElementById "app" =<< (map toNonElementParentNode <<< document) =<< window
   case el of
-    Nothing -> log "Can't find #app element"
-    Just e -> runFn2 React.reactMount e =<< Elmish.construct Counter.def onError
+    Nothing ->
+        log "Can't find #app element"
+    Just e -> do
+        items <- for examples mkItem
+        runFn2 React.reactMount e =<< Elmish.construct (Frame.frame items) onError
 
   where
     onError = log
+
+    mkItem :: Example -> Effect Frame.Item
+    mkItem { title, create } = do
+        view <- create \c -> Elmish.construct c onError
+        pure { title, view }
+
+
+type Example =
+    { title :: String
+    , create :: forall a. (forall m s. ComponentDef Aff m s -> Effect a) -> Effect a
+    }
+
+examples :: Array Example
+examples =
+    [ { title: "Simple counter", create: \f -> f Counter.def }
+    , { title: "Two counters", create: \f -> f TwoCounters.def }
```

Comment:
What does `f` stand for in here? If it’s not relevant, we could just do `(_ Counter.def)`.

Reply:
`f` stands for "callback", but that is not a good term either. No, we can't replace with `_ Counter.def`, because that won't be polymorphic enough.

Reply:
Take a look at the type of `create` to see what `f` is.

Reply:
Thanks for the pointer. Never seen nested `forall` like this so I’ll have to revisit this.